### PR TITLE
Add signature verification/aggregation metrics and panic handlers

### PR DESCRIPTION
**Pull request by Stebalien**
I threw in the panic handlers just to be complete, I've seen too many panics in cryptography code.

### CODE REVIEW

File: blssig/metrics.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	)),
+	verifyAggregate: measurements.Must(meter.Int64Histogram(
+		"f3_blssig_verify_aggregate",
+		metric.WithDescription("Aggregate signatures verified."),
+	)),
+	aggregate: measurements.Must(meter.Int64Histogram(
+		"f3_blssig_aggregate",
+		metric.WithDescription("Signatures aggregated."),
+	)),
+}
```
Comment (Stebalien) [inline blssig/metrics.go:39]: do we need custom buckets here?

File: blssig/metrics.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+		"f3_blssig_verify",
+		metric.WithDescription("Number of signatures verified."),
+	)),
+	verifyAggregate: measurements.Must(meter.Int64Histogram(
+		"f3_blssig_verify_aggregate",
+		metric.WithDescription("Aggregate signatures verified."),
+	)),
+	aggregate: measurements.Must(meter.Int64Histogram(
+		"f3_blssig_aggregate",
+		metric.WithDescription("Signatures aggregated."),
```
Comment (masih) [inline blssig/metrics.go:37]: Extra words for numpties like me?

```suggestion
		metric.WithDescription("Number of signatures aggregated."),
```

File: blssig/metrics.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+		"f3_blssig_decompress_point",
+		metric.WithDescription("Number of times we decompress points."),
+	)),
+	verify: measurements.Must(meter.Int64Counter(
+		"f3_blssig_verify",
+		metric.WithDescription("Number of signatures verified."),
+	)),
+	verifyAggregate: measurements.Must(meter.Int64Histogram(
+		"f3_blssig_verify_aggregate",
+		metric.WithDescription("Aggregate signatures verified."),
```
Comment (masih) [inline blssig/metrics.go:33]:
```suggestion
		metric.WithDescription("Number of aggregate signatures verified."),
```

File: blssig/metrics.go (hunk 1 above)
Comment (masih) [reply blssig/metrics.go:39]: Probably yes, adjusted based on expected network size and quorum? We can refine later if defaults don't make sense.

**Review by masih — APPROVED** [review]

File: blssig/metrics.go (hunk 1 above)
Comment (Stebalien) [reply blssig/metrics.go]: I'll leave the defaults for now.

**Review by Kubuxu — APPROVED** [review]

**Comment by Kubuxu** [comment]
I would prefer if the cache had explicit # cache queries and # cache hits to allow calculating the cache hit ratio.

**Comment by Kubuxu** [comment]
Ahh, I see, you are doing it though a label, I guess it makes sense.

**Pull request merged**